src/sinks/connections/TcpClient.ts

```diff
@@ -24,13 +24,23 @@ export class TcpClient implements ISocketClient {
 
   constructor(endpoint: IEndpoint) {
     this.endpoint = endpoint;
-    this.socket = new net.Socket({ allowHalfOpen: true, writable: false })
+    this.socket = new net.Socket({ allowHalfOpen: true })
       .setEncoding('utf8')
       .setKeepAlive(true)
       .setTimeout(5000) // idle timeout
       .on('timeout', () => this.disconnect('idle timeout'))
       .on('end', () => this.disconnect('end'))
       .on('data', data => LOG('TcpClient received data.', data));
+    this.init.apply(this);
```

Comment:
Having `init()` & `warmup()` adds unclarity on what these methods are doing.

It seems a bit surprising that the socket would start making calls in the constructor, before the object is even fully constructed and returned.
Would making these changes inside `warmup()` be enough to catch this edge case?

Reply:
As discussed, doing in the constructor is not optimal but it's an stablished pattern in this repo. So not a big issue
`firstConnect()` or `initialConnect()` is better, as long as there's a comment explaining why it's needed.